pkg/prompts/prompts.go

```diff
@@ -129,16 +148,58 @@ func NoBlankStringValidator(s string) error {
 	return nil
 }
 
+// Validator for App name
+func appNameValidator(name string) error {
+	if name == "" {
+		return fmt.Errorf("application name cannot be empty")
+	}
+
+	if !unicode.IsLetter(rune(name[0])) && !unicode.IsDigit(rune(name[0])) {
+		return fmt.Errorf("application name must start with a letter or digit")
+	}
+
+	if name[len(name)-1] == '-' || name[len(name)-1] == '_' || name[len(name)-1] == '.' {
+		return fmt.Errorf("application name must end with a letter or digit")
+	}
+
+	for _, r := range name {
+		if !unicode.IsLetter(r) && !unicode.IsDigit(r) && r != '-' && r != '_' && r != '.' {
+			return fmt.Errorf("application name can only contain letters, digits, '-', '_', and '.'")
+		}
+	}
+
+	if len(name) > 63 {
+		return fmt.Errorf("application name cannot be longer than 63 characters")
+	}
+
+	return nil
+}
+
 // RunDefaultableStringPrompt runs a prompt for a string variable, returning the user string input for the prompt
 func RunDefaultableStringPrompt(customPrompt config.BuilderVar, defaultValue string, validate func(string) error, Stdin io.ReadCloser, Stdout io.WriteCloser) (string, error) {
-	var validatorFunc func(string) error
 	if validate == nil {
-		validatorFunc = NoBlankStringValidator
+		validate = NoBlankStringValidator
+	}
+
+	validatorFunc := func(input string) error {
+		// Allow blank inputs because defaults are set later
+		if input == "" {
+			return nil
+		}
+		if customPrompt.Name == "APPNAME" {
+			if err := appNameValidator(input); err != nil {
+				return err
+			}
+		} else {
+			if err := validate(input); err != nil {
+				return err
+			}
+		}
+		return nil
 	}
 
 	defaultString := ""
```

Comment:
can we switch to using the builtin `default` option instead of this empty string method?
https://github.com/manifoldco/promptui/blob/c2e487d3597f59bcf76b24c9e80679740a72212b/prompt.go#L23

Reply:
I was able to do it without adding default in the prompt.